contractcourt/channel_arbitrator.go

```diff
@@ -721,6 +721,15 @@ func (c *ChannelArbitrator) stateStep(triggerHeight uint32,
 		}
 		closeTx = closeSummary.CloseTx
 
+		// Before publishing the closing tx, we persist it to the
```

Comment:
I think there's a gap we're still missing. When we call the `ForceCloseChan` method above, we'll go to mark the channel as borked: https://github.com/lightningnetwork/lnd/blob/master/contractcourt/chain_arbitrator.go#L236. However, we don't specify any additional context here to convey that we're about to force close the channel. As a result, when we come back up, we won't know to trigger the force close again as there's no force close transaction present. 

I'm on the fence w.r.t if we need to care about this scenario atm. This starts to get into the greater issue w.r.t reliably carrying out the DLP steps, since it also applies to all the calls to `MarkBorked` in this method: https://github.com/lightningnetwork/lnd/blob/master/lnwallet/channel.go#L3211 

If we provide the context there, then we'll be able to handle failures at each step of the force close (and also DLP) flow:
  * mark channel action (borked) -> (1) mark commitment broadcast and commit transaction -> (2) broadcast transaction -> (3) update state 

The state written at step 1 should allow us to determine if we need to act (remote party data loss DLP and our local force close), or just halt and wait (local DLP loss). Any auxiliary data such as a channel close summary, or the last chan-reest message sent should also be written at that point. In the DLP case, we need to write the message up front as otherwise when we go to look for it, it won't be present until the channel has been close on chain fully.

Reply:
You're right, there are several cases like this, and feels like they should be handled in a general way.

We could make all the `MarkChanX` methods take an optional close tx, such that we can provide it for the cases where we know of a safe tx to publish. Alternatively we could make the borked state imply that we deem it safe to publish our latest commitment.